src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -720,6 +739,14 @@ private static EntitySourceType GetEntitySourceType(string entityName, Entity en
                     throw new InvalidOperationException($"Target Entity {targetEntityName} should be one of the exposed entities.");
                 }
 
+                if (targetEntity.Source.Object is null)
+                {
+                    throw new DataApiBuilderException(
+                                message: $"Target entity {entityName} does not have a valid source object.",
+                                statusCode: HttpStatusCode.NotAcceptable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
+                }
```

Comment:
Would this  ever be true? Wouldn't the error check you added earlier catch this? `entity.Source.Object is null)` in `GenerateDatabaseObjectForEntities()` which i assume runs first? 

Or is this needed because: yes you do hit the check i just mentioned, but need to continue execution for Dab validate to catch any other errors?

Reply:
this is to catch other errors. Also, the previous one was for source entity, this is for target entity.